neuro_san/client/thinking_file_message_processor.py

```diff
@@ -89,31 +90,57 @@ def write_message(self, response: Dict[str, Any], origin_str: str):
             # There is no real text, but there is a structure. JSON-ify it.
             text = json.dumps(structure, indent=4, sort_keys=True)
 
+        use_origin: str = self._determine_origin_reporting(response, origin_str)
+
+        try:
+            self._write_to_file(origin_str, origin_str, message_type_str, use_origin, text)
+        except OSError as os_error:
+            # For very deep networks we can sometimes get a "File name too long",
+            # which is Error code 63.
+            if os_error.errno == 63:
+                # Retry with a uuid as file name.
+                # If this fails, there's no helping ya.
+                new_uuid: str = str(uuid.uuid4())
+                self._write_to_file(new_uuid, origin_str, message_type_str, use_origin, text)
+            else:
+                raise os_error
+
+    def _write_to_file(self, origin_filename: str, origin_str: str,
+                       message_type_str: str, use_origin: str, text: str):
+
```

Comment:
Break the process of writing to the file out so it can potentially be repeated if an exception comes around.